src/common/consensus-provider/consensus-provider.service.ts

```diff
@@ -294,6 +240,13 @@ export class ConsensusProviderService {
       async (apiURL: string) => this.apiGet(apiURL, this.endpoints.blockInfo(blockId)),
       {
         maxRetries: this.config.get('CL_API_GET_BLOCK_INFO_MAX_RETRIES'),
+        useFallbackOnResolved: (r) => {
+          if (this.workingMode === WorkingMode.Finalized && blockId !== 'head' && r.hasOwnProperty('finalized') && !r.finalized) {
+            this.logger.error(`getBlockInfo: slot [${r.data.message.slot}] is not finalized`);
+            return true;
+          }
```

Comment:
it looks like you can put it to separate func and reuse it in several places

Reply:
I don't think it makes much sense. This similar code is repeated only in 3 places of the `ConsensusProvider` service, and in all these places this code slightly differs from each other. Somewhere we check that the `blockId` / `stateId` is not "head", somewhere we don't. Also, error messages in logs are parametrized by the method name, different parts of this code have different error messages and these error messages have different numbers of output parameters (only slot / state ID and epoch). So, I think that extracting this code into a helper function will rather convolute the code than simplify it.
Please, let me know if you still have another opinion.